---
title: "DE plot"
author: "Abinet Joan"
date: "5/9/2023"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
suppressMessages(library(readxl))
suppressMessages(library(tidyr))
suppressMessages(library(ggplot2))
suppressMessages(library(dplyr))
suppressMessages(library(EnhancedVolcano))
suppressMessages(library(pheatmap))
```

# loading function to save pheatmap as pdf
```{r}
source("save_pheatmap_pdf.R")
```

# Loading perilesional vs intralesional
```{r pressure, echo=FALSE}
file_path <- "/home/qiang/Documents/Travail_cell/Travail_V3/Cécilia_project_2/spatial_transcriptomic/Position_analysis/DE_Intra-vs-Peri-V2.xlsx"

DE_perivsintra <- read_excel(file_path, sheet = "VolcanoPlot", range = "C7:H19969")
```

## making volcano plot
```{r}
DE_perivsintra$padj <- as.numeric(DE_perivsintra$`Adjusted pvalue`)

EnhancedVolcano(DE_perivsintra, DE_perivsintra$`Target name`, x = "Log2", y = "padj", title= "DE-perilesional vs intralesional", FCcutoff = 1, pCutoff = 0.05, labSize = 0.1, ylim = c(0,4), xlim = c(-5.5,3), gridlines.major= F, gridlines.minor = F)
#ggsave("volcanoplot/Volcano_perivsintraV2.pdf", width = 10, height = 10)
```
## Gene for Heatmap (in red in the volcanoplot)
```{r}
gene_significant <- subset(DE_perivsintra, (Log2 > 1 | Log2< -1) & padj <=0.05)
gene_significant <- gene_significant[order(gene_significant$Log2, decreasing = F),]
gene_significant <- gene_significant[1:30,]
gene_significant <- gene_significant$`Target name`
```

## load the annotation file from Q3 - publications
```{r}
colData <- read_excel("../Data_Thomas/Q3 - publication.xlsx", sheet = "SegmentProperties")
colData <- as.data.frame(colData)
rownames(colData) <- colData$SegmentDisplayName

colData[8,25] <- "Perilesional"
```


## Load the normalised count matrix and making the heatmap
```{r}
Q3_publication <- read_excel("../Data_Thomas/Q3 - publication.xlsx", sheet = "TargetCountMatrix")
Q3_publication <- as.data.frame(Q3_publication)
rownames(Q3_publication) <- Q3_publication$TargetName
Q3_publication$TargetName <- NULL

rld <- log2(Q3_publication)

df <- as.data.frame(colData[,c("SlideName","Group")])

Heatmap_perivsintra <- pheatmap(rld[gene_significant,], cluster_rows=FALSE, show_rownames=T,
         cluster_cols=T, annotation_col=df, scale = "row", main = "Genes differentially expressed in Perilesional over intralesionnal" )

#save_pheatmap_pdf(Heatmap_perivsintra, "Heatmap/Heatmap_perivsintraV2.pdf", width = 10, height = 10)
```


# Loading unaffected vs intralesionnal
```{r}
DE_UnaffectedvsIntra <- read_excel("DE-Intra-vs-Unaffected-V2.xlsx", sheet = "VolcanoPlot", range = "C7:H19969")
DE_UnaffectedvsIntra$padj <- as.numeric(DE_UnaffectedvsIntra$`Adjusted pvalue`)

EnhancedVolcano(DE_UnaffectedvsIntra, DE_UnaffectedvsIntra$`Target name`, x = "Log2", y = "padj", title= "DE-unaffected vs intralesionnal", FCcutoff = 1, pCutoff = 0.05, labSize = 0.1, ylim = c(0,4), xlim = c(-5.5,3), gridlines.major= F, gridlines.minor = F)
#ggsave("volcanoplot/Volcano_UnaffectedvsIntraV2.pdf", width = 10, height = 10)
```
## Heatmap !! je ne prends pas tous les gènes significatif
```{r}
gene_significant <- subset(DE_UnaffectedvsIntra, (Log2 > 1 | Log2< -1) & padj <=0.05)
gene_significant_pos <- gene_significant[gene_significant$Log2 >=1,]
gene_significant_neg <- gene_significant[gene_significant$Log2 <= -1,]
gene_significant_neg <- gene_significant_neg[order(gene_significant_neg$Log2),]
gene_toplot <- c(gene_significant_neg$`Target name`[1:20], gene_significant_pos$`Target name`)


Heatmap_UnaffectedvsIntra <- pheatmap(rld[gene_toplot,], cluster_rows=FALSE, show_rownames=T,
         cluster_cols=T, annotation_col=df, scale = "row", main = "Genes differentially expressed in unaffected over intralesionnal" )
#save_pheatmap_pdf(Heatmap_UnaffectedvsIntra, "Heatmap/Heatmap_UnaffectedvsIntra.pdf", width = 10, height = 10)
```



# DE unaffected vs perilesional
```{r}
DE_UnaffectedvsPeri <- read_excel("DE-Peri-vs-Unaffected-V2.xlsx", sheet = "VolcanoPlot", range = "C7:H19969")
DE_UnaffectedvsPeri$padj <- as.numeric(DE_UnaffectedvsPeri$`Adjusted pvalue`)

EnhancedVolcano(DE_UnaffectedvsPeri, DE_UnaffectedvsPeri$`Target name`, x = "Log2", y = "padj", title= "DE-unaffected vs perilesional", FCcutoff = 1, pCutoff = 0.05, labSize = 0.1, ylim = c(0,4), xlim = c(-5.5,3), gridlines.major= F, gridlines.minor = F)
#ggsave("volcanoplot/Volcano_UnaffectedvsPeri.pdf", width = 10, height = 10)
```

## Heatmap !! je ne prends pas tous les gènes significatif
```{r}
gene_significant <- subset(DE_UnaffectedvsPeri, (Log2 > 1 | Log2< -1) & padj <=0.05)
gene_significant_pos <- gene_significant[gene_significant$Log2 >=1,]
gene_significant_pos <- gene_significant_pos[order(gene_significant_pos$Log2, decreasing = T),]
gene_significant_neg <- gene_significant[gene_significant$Log2 <= -1,]
gene_significant_neg <- gene_significant_neg[order(gene_significant_neg$Log2),]

gene_toplot <- c(gene_significant_neg$`Target name`[1:20], gene_significant_pos$`Target name`[1:20])


Heatmap_UnaffectedvsPeri <- pheatmap(rld[gene_toplot,], cluster_rows=FALSE, show_rownames=T,
         cluster_cols=T, annotation_col=df, scale = "row", main = "Genes differentially expressed in unaffected over perilesional" )
#save_pheatmap_pdf(Heatmap_UnaffectedvsPeri, "Heatmap/Heatmap_UnaffectedvsPeri.pdf", width = 10, height = 10)
```

# DE unaffected vs Healthy
```{r}
DE_UnaffectedvsHealthy <- read_excel("DE-Unaffected-vs-Healthy-V2.xlsx", sheet = "VolcanoPlot", range = "C7:H19969")
DE_UnaffectedvsHealthy$padj <- as.numeric(DE_UnaffectedvsHealthy$`Adjusted pvalue`)

EnhancedVolcano(DE_UnaffectedvsHealthy, DE_UnaffectedvsHealthy$`Target name`, x = "Log2", y = "padj", title= "DE-unaffected vs Healthy", FCcutoff = 1, pCutoff = 0.05, labSize = 0.1, ylim = c(0,4), gridlines.major= F, gridlines.minor = F)
#ggsave("volcanoplot/Volcano_UnaffectedvsHealthy.pdf", width = 15, height = 15)
```

# Heatmap (no gene significant in healthy condition)
```{r}
gene_significant <- subset(DE_UnaffectedvsHealthy, (Log2 > 1 | Log2< -1) & padj <=0.05)
gene_significant_neg <- gene_significant[gene_significant$Log2 <= -1,]
gene_significant_neg <- gene_significant_neg[order(gene_significant_neg$Log2),]

gene_toplot <- c(gene_significant_neg$`Target name`[1:20])

Heatmap_UnaffectedvsHealthy <- pheatmap(rld[gene_toplot,], cluster_rows=FALSE, show_rownames=T,
         cluster_cols=T, annotation_col=df, scale = "row", main = "Genes differentially expressed in unaffected over Healthy" )
save_pheatmap_pdf(Heatmap_UnaffectedvsHealthy, "Heatmap/Heatmap_UnaffectedvsHealthy.pdf", width = 10, height = 10)
```

# Heatmap with all comparison
```{r}
List_DE <- list(DE_perivsintra, DE_UnaffectedvsIntra, DE_UnaffectedvsPeri, DE_UnaffectedvsHealthy)

# select le top 10 gènes significatif de chaque groupe
heatmap_genes <- list()
i <- 1 
for (result_genes in List_DE) {
  
  result_genes <- subset(result_genes, padj <=0.05)
  genes_pos <- result_genes[result_genes$Log2 >=0,]
  genes_pos <- genes_pos[order(genes_pos$Log2, decreasing = T),]
  heatmap_genes[[i]] <- genes_pos$`Target name`[1:10]
  i <- i+1
  
  genes_neg <- result_genes[result_genes$Log2 < 0,]
  genes_neg <- genes_neg[order(genes_neg$Log2),]
  heatmap_genes[[i]] <- genes_neg$`Target name`[1:10]
  i <- i+1
}

heatmap_genes <- c(heatmap_genes[[7]], heatmap_genes[[4]], heatmap_genes[[6]], heatmap_genes[[8]], heatmap_genes[[2]], heatmap_genes[[5]], heatmap_genes[[1]], heatmap_genes[[3]])

Heatmap_allcondition <- pheatmap(rld[heatmap_genes,c(1,5,13,17,9,12,18,23,2:4,6:8,10,11,14,15,24,16,19:22)],
                                 cluster_rows=FALSE, show_rownames=T, cluster_cols=F, annotation_col= df, scale = "row", main = "Genes differentially expressed in all condition" )

save_pheatmap_pdf(Heatmap_allcondition, "Heatmap/Heatmap_allcondition.pdf", width = 10, height = 15)
```

