---
title: "Preparing files for GSEA analysis"
author: "Abinet Joan"
date: "`r format (Sys.time(), format='%Y-%m-%d %H:%M:%S %z')`"
output:
  pdf_document:
    toc: true
    toc_depth: 2
header-includes:
  - |
    ```{=latex}
    \usepackage{fvextra}
    \DefineVerbatimEnvironment{Highlighting}{Verbatim}{
      breaksymbolleft={}, 
      showspaces = false,
      showtabs = false,
      breaklines,
      commandchars=\\\{\}
    }
    ```
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Loading packages
```{r cars}
suppressMessages(library(readxl))
suppressMessages(library(tidyr))
suppressMessages(library(ggplot2))
suppressMessages(library(dplyr))
```

# Preparing gct and cls files for GSEA
The GSEA require 3 files that need to be generated from the sequencing data: A gct files describing the gene expression matrix and a cls files describing the metadata for each cells

A gmt files containing Molecular Signatures Database is directly downloaded from internet.

## Perilesionnal vs intralesional

```{r}
Q3 <- read_excel("../../Data_DSPDA/Q3 - publication - V2.xlsx", sheet = "TargetCountMatrix")
Q3 <- as.data.frame(Q3)
rownames(Q3) <- Q3$TargetName
Q3$TargetName <- NULL

Q3 <- Q3[,-c(1,5,9,12,13,17,18,23)]
```


```{r}
# gct files
avg = as.data.frame(Q3)
df = data.frame(NAME=rownames(avg), Description="NA", stringsAsFactors = F)
df = cbind(df, avg)
df = rbind(c("#1.2", rep(NA, ncol(df)-1)), c(nrow(df), ncol(df)-2, rep(NA, ncol(df)-2)), colnames(df), df)
#write.table(df, file = "exprMat.gct", sep = "\t", row.names = F, col.names = F, na = "", quote = F)
```

```{r}
# cls files
annotation_sample <- c(rep("perilesional",10), rep("intralesional", 5), "perilesional")

no_tot = ncol(avg)
no_classes = c("perilesional","intralesional")
df = data.frame(matrix(ncol=no_tot))
df[1,] = c(no_tot, length(no_classes), 1, rep(NA, no_tot - 3))
line2 = "#"
line3 = as.vector(annotation_sample)
for (i in 1:length(no_classes)){
  line2 = c(line2, as.vector(unique(annotation_sample)[i]))
}
line2 = c(line2, rep(NA, no_tot - length(line2)))
df = rbind(df, line2, line3)
#write.table(df, file = "MetaData.cls", sep = "\t", row.names = F, col.names = F, na = "", quote = F)
```

## Perilesionnal vs unaffected
```{r}
Q3 <- read_excel("../../Data_DSPDA/Q3 - publication - V2.xlsx", sheet = "TargetCountMatrix")
Q3 <- as.data.frame(Q3)
rownames(Q3) <- Q3$TargetName
Q3$TargetName <- NULL

Q3 <- Q3[,-c(1,5,13,17,16,19,20,21,22)]
```

```{r}
# gct files
avg = as.data.frame(Q3)
df = data.frame(NAME=rownames(avg), Description="NA", stringsAsFactors = F)
df = cbind(df, avg)
df = rbind(c("#1.2", rep(NA, ncol(df)-1)), c(nrow(df), ncol(df)-2, rep(NA, ncol(df)-2)), colnames(df), df)
write.table(df, file = "PerivsUnaffected/exprMat.gct", sep = "\t", row.names = F, col.names = F, na = "", quote = F)
```

```{r}
# cls files
annotation_sample <- c(rep("perilesional",6), "Unaffected", rep("perilesional",2),"Unaffected", rep("perilesional",2), rep("Unaffected",2), "perilesional")

no_tot = ncol(avg)
no_classes = c("perilesional","intralesional")
df = data.frame(matrix(ncol=no_tot))
df[1,] = c(no_tot, length(no_classes), 1, rep(NA, no_tot - 3))
line2 = "#"
line3 = as.vector(annotation_sample)
for (i in 1:length(no_classes)){
  line2 = c(line2, as.vector(unique(annotation_sample)[i]))
}
line2 = c(line2, rep(NA, no_tot - length(line2)))
df = rbind(df, line2, line3)
# write.table(df, file = "PerivsUnaffected/MetaData.cls", sep = "\t", row.names = F, col.names = F, na = "", quote = F)
```



## Intralesionnal vs unaffected
```{r}
Q3 <- read_excel("../../Data_DSPDA/Q3 - publication - V2.xlsx", sheet = "TargetCountMatrix")
Q3 <- as.data.frame(Q3)
rownames(Q3) <- Q3$TargetName
Q3$TargetName <- NULL

Q3 <- Q3[,c(9,12,16,18,19,20,21,22,23)]
```

```{r}
# gct files
avg = as.data.frame(Q3)
df = data.frame(NAME=rownames(avg), Description="NA", stringsAsFactors = F)
df = cbind(df, avg)
df = rbind(c("#1.2", rep(NA, ncol(df)-1)), c(nrow(df), ncol(df)-2, rep(NA, ncol(df)-2)), colnames(df), df)
# write.table(df, file = "IntravsUnaffected/exprMat.gct", sep = "\t", row.names = F, col.names = F, na = "", quote = F)
```

```{r}
# cls files
annotation_sample <- c(rep("Unaffected",2), "intralesionnal","Unaffected", rep("intralesionnal",4), "Unaffected")

no_tot = ncol(avg)
no_classes = c("Unaffected","intralesionnal")
df = data.frame(matrix(ncol=no_tot))
df[1,] = c(no_tot, length(no_classes), 1, rep(NA, no_tot - 3))
line2 = "#"
line3 = as.vector(annotation_sample)
for (i in 1:length(no_classes)){
  line2 = c(line2, as.vector(unique(annotation_sample)[i]))
}
line2 = c(line2, rep(NA, no_tot - length(line2)))
df = rbind(df, line2, line3)
#write.table(df, file = "IntravsUnaffected/MetaData.cls", sep = "\t", row.names = F, col.names = F, na = "", quote = F)
```


## Intralesionnal vs unaffected
```{r}
Q3 <- read_excel("../../Data_DSPDA/Q3 - publication - V2.xlsx", sheet = "TargetCountMatrix")
Q3 <- as.data.frame(Q3)
rownames(Q3) <- Q3$TargetName
Q3$TargetName <- NULL

Q3 <- Q3[,c(1,5,9,12,13,17,18,23)]
```

```{r}
# gct files
avg = as.data.frame(Q3)
df = data.frame(NAME=rownames(avg), Description="NA", stringsAsFactors = F)
df = cbind(df, avg)
df = rbind(c("#1.2", rep(NA, ncol(df)-1)), c(nrow(df), ncol(df)-2, rep(NA, ncol(df)-2)), colnames(df), df)
#write.table(df, file = "unaffectedvshealthy/exprMat.gct", sep = "\t", row.names = F, col.names = F, na = "", quote = F)
```

```{r}
# cls files
annotation_sample <- c(rep("Healthy",2), rep("Unaffected",2), rep("Healthy",2), rep("Unaffected",2))

no_tot = ncol(avg)
no_classes = c("Unaffected","Healthy")
df = data.frame(matrix(ncol=no_tot))
df[1,] = c(no_tot, length(no_classes), 1, rep(NA, no_tot - 3))
line2 = "#"
line3 = as.vector(annotation_sample)
for (i in 1:length(no_classes)){
  line2 = c(line2, as.vector(unique(annotation_sample)[i]))
}
line2 = c(line2, rep(NA, no_tot - length(line2)))
df = rbind(df, line2, line3)
#write.table(df, file = "unaffectedvshealthy/MetaData.cls", sep = "\t", row.names = F, col.names = F, na = "", quote = F)
```

Then GSEA will be run and the results can be found in the directory 7.2-GSEA/output

The Dotplot are generated by the code in 7.2-GSEA/Plots/Creating_Dotplot.rmd

```{r}
sessionInfo()
```

